iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/AbstractArmTreeImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.iac.arm.tree.api.ArmTree;
+import org.sonar.iac.common.api.tree.HasTextRange;
+import org.sonar.iac.common.api.tree.impl.TextRange;
+import org.sonar.iac.common.api.tree.impl.TextRanges;
+
+public abstract class AbstractArmTreeImpl implements ArmTree {
+
+  protected TextRange textRange;
+  protected ArmTree parent;
+
+  @Override
+  public TextRange textRange() {
+    if (textRange == null) {
+      List<TextRange> childRanges = children().stream().map(HasTextRange::textRange).collect(Collectors.toList());
+      textRange = TextRanges.merge(childRanges);
+    }
+    return textRange;
+  }
```

Comment:
I thought about `default` method in `Tree` interface, but then it needs to be recalculated every time because fields in Java interfaces are `final`. I'm not sure if introducing a next level in our already big class hierarchy to save 8 lines of code is a good idea 🤔  The benefit is small in comparison to complexity. 
Anyway if still needed it's better to handle it in separate PR